### DISCUSSION TITLE
Issue with navigation to [id]/page.tsx with Link

### DISCUSSION THREAD

**Discussion by AndonMitev** [vercel/next.js#52634] [App Router]
This is the app structure:

```
app/
     games/
           page.tsx
           [id]/
                 page.tsx
```

And trying to reach /games/id somehow does not work :(

This is how navigation is done:

```
 <Link href={`/games/55`}>Game Details</Link>
```

instead of landing in `[id]/page.tsx` user is sent to `app/page.tsx`

and by the way this is happening only in production on dev env is working as expected there is some delay until page is loaded that is issue as well but looks like only dev trouble

```
    "next": "13.4.5",
     "eslint-config-next": "13.4.5",
```

**Comment by icyJoseph** [comment]
How are you rendering `games/[id]/page.tsx`? Are you using `export` mode?

**Reply by AndonMitev** [reply]
```
export default function Game() {
  return (
    <div className="grid grid-cols-[300px_1fr_300px] mt-2 gap-4">
      <p className="text-white">Game Details</p>
    </div>
  );
}
```

**Reply by AndonMitev** [reply]
PS: Tried with `13.4.9` - same issue

**Reply by icyJoseph** [reply]
Mmm really that's all the logic in game details? This is really strange. Do you have a repository I could take a look at?

**Reply by AndonMitev** [reply]
![image](https://github.com/vercel/next.js/assets/32274987/0d28ec51-3735-44d7-8fca-aea16b9dd1c9)


Here its not open source sorry

**Reply by AndonMitev** [reply]
![image](https://github.com/vercel/next.js/assets/32274987/06b1674f-ea93-4039-b576-78ccf49c0cd4)
Here is the navigation with link

**Reply by icyJoseph** [reply]
what happens if you move things to `gamess` with double s at the end, and work with that instead? 

My suspicion being that maybe there's some redirect that's messing things up - 

but why only in prod... Mmm the plot thickens

**Reply by AndonMitev** [reply]
Something that i can spot is that when i click on the link i see how the whole application is reloaded on production ( i do have some async functions to determine if user is logged or not ) and when clicking on the link i can see how things are being reloaded:

Something else interesting is that i do have 2 bars:

in `layout.tsx`

```
  <body id="home-body" className={inter.className}>
 <div className=" fixed w-full h-[140px] top-0 left-0 z-10 bg-gray-800">
                <MainNav />
                <FavoriteGamesNavbar />
                <RouteGuard />
              </div>
              <main id="home-root" className=" mt-[140px] mx-4 flex flex-col gap-4 relative flex-1 h-full">
                {children}
              </main>
</body>
```

MainNav is working as expected FavoritesGamesNavbar is doing the trouble on production:

MainNav:

![image](https://github.com/vercel/next.js/assets/32274987/49d6816e-cd68-4ae0-b959-96ab10bd0bc7)


FavNav:
![image](https://github.com/vercel/next.js/assets/32274987/6c696d59-4d10-43bd-b53a-36cbea7fa9ce)


And when link is clicked in FavNav, feels like whole app is being reloaded

idk if that makes sense

**Reply by AndonMitev** [reply]
Ok more info:

I have moved `/games/55` to be called in MainNav that is working, once when start calling this route app changes behaviour:
- reload from scratch
- not able to open the route

**Reply by AndonMitev** [reply]
@icyJoseph with `gamess` - same thing `/gamess/[id]/page.tsx` is not opened but `/gamess/page.tsx` there is no problem with it

**Reply by icyJoseph** [reply]
Mmm - alright, but it looks like it is something with your structure, but I can't tell what it is. I wonder if it is a bug with the `(auth)` group, but that sounds really strange.

Sorry mate, without a reproduction demo, I can't do much :/ maybe I am even distracting you

**Reply by AndonMitev** [reply]
I see bro, thanks a lot, here created example repo which again is not working on my side:

https://github.com/AndonMitev/next-router-id-example

but starting to think that might be coming from tauri actually

**Reply by AndonMitev** [reply]
@icyJoseph can you share if structure make sense or there can be any issue

**Reply by AndonMitev** [reply]
Also I do use this:

```
next.config.js

/** @type {import('next').NextConfig} */
const nextConfig = {
  output: 'export',
  images: {
    loader: 'custom',
    loaderFile: './lib/clodinaryLoader.ts',
  },
  webpack: (config, { isServer }) => {
    if (!isServer) {
      config.externals.push({
        bufferutil: 'bufferutil',
        lokijs: 'lokijs',
        encoding: 'encoding',
        'utf-8-validate': 'utf-8-validate',
      });
    }

    return config;
  },
};

module.exports = nextConfig;

```

and i saw this thread as well:
https://github.com/vercel/next.js/issues/48022

**Reply by icyJoseph** [reply]
Aha, so you were using `export`, yeah dynamic routes with export do a 404 IIRC.

Could it be that the 404 is routed to root by Tauri?
